wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/spi/environment/InternalEnvironmentFactory.java

```diff
@@ -42,22 +44,45 @@
  *   envFactories.addBinding("recipe_type_1").to(SubclassOfInternalEnvironmentFactory.class);
  * </pre>
  *
+ * defaultMachineMaxMemorySizeAttribute defines default machine memory limit
```

Comment:
I think it would be better to rewrite this doc like in the following way:
```
Implementation should add memory-related attributes into machine config if they are not specified. The values should be fetched from workspace recipe or default values from protected fields should be used.
There are two memory-related attributes: 
- memory limit - ....
- memory request - ...
```
It's not ready to copy/paste one but it shows the idea to define the contract of abstract class but not just protected fields.
WDYT?